packages/core/src/directory.ts

```diff
@@ -0,0 +1,263 @@
+                name: group.name,
+            });
+
+            org.groups.push(newGroup);
+
+            await this.server.updateMetaData(org);
+            await this.server.storage.save(org);
+
+            return newGroup;
+        }
+    }
+
+    async groupUpdated(groupChanges: DirectoryGroupChanges, orgId: string, groupId: string) {
+        const org = (orgId && (await this.server.storage.get(Org, orgId))) || null;
+        if (org && org.directory.syncGroups) {
+            const existingGroup = org.groups.find((orgGroup) => orgGroup.name === groupId);
+
+            if (!existingGroup) {
+                return;
+            }
+
+            if (groupChanges.newName) {
+                existingGroup.name = groupChanges.newName;
+            }
+
+            if (groupChanges.memberIdsToAdd) {
+                for (const memberIdToAdd of groupChanges.memberIdsToAdd) {
+                    let existingOrgMember: OrgMember | null = null;
+                    for (const member of org.members) {
+                        if (existingOrgMember) {
+                            continue;
+                        }
+
+                        const provisioningId = await getIdFromEmail(member.email);
+
+                        if (
+                            member.accountId === memberIdToAdd ||
+                            member.id === memberIdToAdd ||
+                            provisioningId === memberIdToAdd
+                        ) {
+                            existingOrgMember = member;
+                        }
+                    }
+
+                    if (existingOrgMember) {
+                        existingGroup.members.push(existingOrgMember);
+                    }
+                }
+            }
+
+            if (groupChanges.memberIdsToRemove) {
+                for (const memberIdToRemove of groupChanges.memberIdsToRemove) {
+                    let existingOrgMember: OrgMember | null = null;
+                    for (const member of org.members) {
+                        if (existingOrgMember) {
+                            continue;
+                        }
+
+                        const provisioningId = await getIdFromEmail(member.email);
+
+                        if (
+                            member.accountId === memberIdToRemove ||
+                            member.id === memberIdToRemove ||
+                            provisioningId === memberIdToRemove
+                        ) {
+                            existingOrgMember = member;
+                        }
+                    }
+
+                    if (existingOrgMember) {
+                        const existingMemberIndex = existingGroup.members.findIndex(
+                            ({ email }) => existingOrgMember!.email === email
+                        );
+                        if (existingMemberIndex !== -1) {
+                            existingGroup.members.splice(existingMemberIndex, 1);
```

Comment:
@MaKleSoft any idea why this isn't enough to persist a group membership removal? Adding by mutating this array works, but removing doesn't persist. It's the final piece to start testing with real AD.

Reply:
Hm, no idea. I can try debugging this once we have a test setup though.

Reply:
So you're saying it should be enough to simply update the `members` array and save the org, right?